Automattic/woocommerce-payments
# Use the Stripe Exchange Rate in Analytics

Fixes #2581 

#### Changes proposed in this Pull Request









- Use the Stripe Exchange rate to calculate order values for analytics, with the order exchange rate (which was used to calculate the front-end display) used as a fallback if this is not available.

#### Testing instructions





* Checkout this branch.
* Make an order in your checkout using a multi-currency currency.
* Go to `Tools > Scheduled Actions` and run the pending `wc-admin_import_orders` action associated with the order ID you just created (note: this may have already been run automatically).
* Verify that the order has a `_wcpay_multi_currency_stripe_exchange_rate` meta key set, either by checking the database in the post meta table, or manually checking using a code snippet such as:
```php
$order = wc_get_order( <order_id> );
$order->get_meta( '_wcpay_multi_currency_stripe_exchange_rate' );
```

* Comment out the following lines in `class-wc-payment-gateway-wcpay.php` (lines 1118 and 1119):
```php
$order->update_meta_data( '_wcpay_multi_currency_stripe_exchange_rate', $exchange_rate );
$order->save_meta_data();
```
* Make an order in your checkout of the same value as the previous order, using the same multi-currency currency.
* Go to `Tools > Scheduled Actions` and run the pending `wc-admin_import_orders` action associated with the order ID you just created (note: this may have already been run automatically).
* Verify that the order does **not** have a  `_wcpay_multi_currency_stripe_exchange_rate` meta key set.
* Check the `wp_wc_order_stats` table in your database. You should see the two orders, both should have been converted into your store's default currency. The first order should have been converted using the Stripe exchange rate, the second one should have been converted using the frontend conversion rate (the rate we see in Multi-Currency).
* Go to the analytics pages, filtering to show today's data, verify that both orders appear and that the data looks sensible. Check each analytics page to ensure it loads properly (note that some pages may not have data, e.g. if you haven't used any coupons, the coupons page will have no data, but it should still load without any error message and display an empty table).


-------------------
